docs/commands/cp.md

```diff
@@ -0,0 +1,129 @@
+# cp
+
+Copy storage files and directories between cloud and local storage.
+
+## Synopsis
+
+```usage
+usage: datachain cp [-h] [-v] [-q] [-r] [--team TEAM]
+    [-s] [--update]
+    source_path destination_path
+```
+
+## Description
+
+This command copies files and directories between local and/or remote storage. This uses the credentials in your system by default or can use the cloud authentication from Studio.
+
+## Arguments
+
+* `source_path` - Path to the source file or directory to copy
+* `destination_path` - Path to the destination file or directory to copy to
+
+## Options
+
+* `-r`, `-R`, `--recursive` - Copy directories recursively
+* `--team TEAM` - Team name to use the credentials from.
+* `-s`, `--studio-cloud-auth` - Use credentials from Studio for cloud operations (Default: False)
+* `--update` - Update cached list of files for the source when downloading from cloud using local credentials.
+* `-h`, `--help` - Show the help message and exit
+* `-v`, `--verbose` - Be verbose
+* `-q`, `--quiet` - Be quiet
+
+## Copy Operations
+
+The command supports two main modes of operation:
+
+### Default Mode
+By default, the command operates directly with clouds using credentials in ypur system, supporting various copy scenarios between local and remote storage.
+
+### Studio Cloud Auth Mode
+When using `-s` or `--studio-cloud-auth` flag, the command uses credentials from Studio for cloud operations. This mode provides enhanced authentication and access control for cloud storage operations.
+
+## Supported Storage Protocols
+
+The command supports the following storage protocols:
+- **Local file system**: Direct paths (e.g., `/path/to/directory` or `./relative/path`)
+- **AWS S3**: `s3://bucket-name/path`
+- **Google Cloud Storage**: `gs://bucket-name/path`
+- **Azure Blob Storage**: `az://container-name/path`
+
+## Examples
+
+The command automatically determines the operation type based on the source and destination protocols:
+
+### Local to Local
+
+**Operation**: Direct local file system copy
+- Uses the local filesystem's native copy operation
+- Fastest operation as no network transfer is involved
+- Supports both files and directories
+
+```bash
+datachain cp /path/to/source/file.py /path/to/destination/file.py
+datachain cp -r /path/to/source/directory /path/to/destination/directory
+```
+
+### Local to Remote
```

Comment:
I don't think we need so many examples ...

Reply:
I think otherwise, There are exactly 4 different types of examples to show what are supported. Within each type, we have file and folders each using local and studio authentication. 

If you think otherwise, let me know which examples to keep.

Reply:
just keep 2-3 examples, command and options should be more or less self descriptive

or make examples more comprehensive - e.g. do Studio auth, etc ...

again, otherwise we just repeat the option descritptions

Reply:
> just keep 2-3 examples, command and options should be more or less self descriptive

Which ones? Asking before making any changes to avoid back and forth.